src/lib.rs

```diff
@@ -376,38 +360,62 @@ impl<R: gimli::Reader> Context<R> {
     /// to the innermost inline function.  Subsequent frames contain the caller and call
     /// location, until an non-inline caller is reached.
     pub fn find_frames(&self, probe: u64) -> Result<FrameIter<R>, Error> {
-        let (unit_id, loc, funcs) =
-            match self.find_unit_id_and_address(probe) {
-                Some((unit_id, address)) => {
-                    let unit = &self.units[unit_id];
-                    let loc = unit.find_location(probe, &self.sections)?;
-                    let functions = unit.parse_functions(&self.sections, &self.units)?;
-                    let mut res = maybe_small::Vec::new();
-                    let mut function_index = functions.addresses[address].function;
-                    loop {
-                        let function = &functions.functions[function_index];
-                        res.push(function);
-                        if let Some(inlined) = function.inlined.iter().find(|inlined| {
-                            probe >= inlined.range.begin && probe < inlined.range.end
-                        }) {
-                            function_index = inlined.function;
-                            continue;
-                        } else {
-                            break;
-                        }
-                    }
-                    (unit_id, loc, res)
+        let (unit, address) = match self.find_unit_and_address(probe) {
+            Some(x) => x,
+            None => return Ok(FrameIter(FrameIterState::Empty)),
+        };
+
+        let loc = unit.find_location(probe, &self.sections)?;
+        let functions = unit.parse_functions(&self.sections)?;
+        let function_index = functions.addresses[address].function;
+        let function = &functions.functions[function_index];
+        let function = function
+            .1
+            .borrow_with(|| Function::parse(function.0, &unit.dw_unit, &self.sections, &self.units))
+            .as_ref()
+            .map_err(Error::clone)?;
+
+        // Build the list of inlined functions that contain probe.
+        // funcs is ordered from outside to inside.
+        // The indexes are into the function.inline_functions vec.
```

Comment:
I'm not sure which "indexes" this comment refers to. Looking at my PR, the comment looked just as lost in it. I think it can be removed.

Reply:
I think I changed the code to return a reference instead of the index.